Describe Authorization scopes

bajtos opened the pull request:
Documentation for the new feature to be added by https://github.com/strongloop/loopback/pull/3313.

Connect to strongloop/loopback#382

bajtos committed:
Describe Authorization scopes

bajtos commented:
(This pull request must not be landed before the change in LoopBack is finished.)

crandmck commented:
Actually, the doc PR should not be landed until the code is landed AND published to npm.

bajtos commented:
I have reworked the pull request to match what has been landed by strongloop/loopback#3313.

@ebarault @crandmck could you please review again?

crandmck reviewed (changes requested):
I went ahead and made wording and grammar edits directly in the branch, since that seemed most efficient.

IIUC, we should also update https://loopback.io/doc/en/lb3/Model-definition-JSON-file.html#methods since this feature will now be implemented.  Is that right?
If so, please add to this PR--or if you prefer I can take a stab at it next week.
Again, if I'm understanding, this reference info should largely duplicate what's in https://loopback.io/doc/en/lb3/Remote-methods.html#options.   If so, then instead of 
actually duplicating these tables, we can move them into the `_includes/content` and
single-source them so we only have to maintain them there.

bajtos commented on pages/en/lb3/Controlling-data-access.md:
```diff
@@ -253,45 +253,41 @@ Assuming the following ACL rules are defined:
-The scopes are defined together with other remoting metadata via the new
-setting `accessScopes`. The user can invoke the remote method as long as their access token
-is granted at least one of the scopes listed in `accessScopes` array.
+The new model property `accessScopes` defines scopes as part of remoting metadata.
```
@crandmck 
`accessScopes` is not a model property, it is a new property in remote method metadata. My proposal:

```diff
- The new model property `accessScopes` defines scopes as part of remoting metadata.
+ The new setting `accessScopes` defines scopes as part of remoting metadata.
```

crandmck replied:
IIUC, it's a value set in the model JSON file, which is what I meant by a "model property".  Apparently that is confusing, but "setting" doesn't really convey any specific meaning.  

Perhaps it would be clearer to say 
> The `methods.accessScopes` property in the model JSON file defines...

bajtos replied:
Done in ac92730

bajtos commented:
@crandmck thank you for the edits.

> https://loopback.io/doc/en/lb3/Remote-methods.html#options

Good catch, `accessScopes` should be added to that table. I'll do that ASAP.

> IIUC, we should also update https://loopback.io/doc/en/lb3/Model-definition-JSON-file.html#methods since this feature will now be implemented. Is that right?

I am confused. The current doc pages says that remote methods cannot be configured via JSON files yet, but that hasn't been true for months. So yes, we should fix that section and I think that fix is independent of the changes I am making here. I am proposing to leave that out of scope of this pull request.

> Again, if I'm understanding, this reference info should largely duplicate what's in https://loopback.io/doc/en/lb3/Remote-methods.html#options. If so, then instead of
actually duplicating these tables, we can move them into the _includes/content and
single-source them so we only have to maintain them there.

Ditto, I am proposing to leave `_includes/content` out of scope of this patch.

bajtos commented:
Updated, PTAL.

crandmck commented:
> The current doc pages says that remote methods cannot be configured via JSON files yet, but that hasn't been true for months. So yes, we should fix that section and I think that fix is independent of the changes I am making here. I am proposing to leave that out of scope of this pull request.

No one told me, created an issue, or opened a doc PR, so apparently it was not documented (unless I'm missing something).  It's closely related to this PR, but we can do it separately if you prefer.
Added #352.

With [one small edit](https://github.com/strongloop/loopback.io/pull/338#discussion_r110962379) noted above I think this is ready to land, WHEN the change is published to npm.  We shouldn't land this until that time, though.

crandmck commented:
Note: https://github.com/strongloop/loopback.io/commit/c3059946e7673214dc97f2487aa09ed83414c849 will require this to be rebased, because I added the doc for remote method config via JSON per #352.

bajtos commented:
> Note: c305994 will require this to be rebased, because I added the doc for remote method config via JSON per #352.

Rebased, I moved the newly-added remote method option "accessScopes" to the new place you created.

>  I think this is ready to land, WHEN the change is published to npm. We shouldn't land this until that time, though.

Sounds good. I'd like to land https://github.com/strongloop/loopback/pull/3350 before making a new loopback 3.x release.

bajtos commented:
All required changes were landed and released in `loopback@3.7.0`. I'm going to rebase and land this patch.
